src/plugins/output_format/jsonfmt.h

```diff
@@ -293,6 +293,8 @@ class DataPrinter<T, std::enable_if_t<is_iterable<T>::value&& has_mapped_type<T>
                 os << sep;
 
             bool printed_rest = DataPrinter<T>::print_data(os, sep, args...);
+            if (!printed_rest)
```

Comment:
I would appreciate an inline comment explaining why we have this check here.

Reply:
Added a comment and a small fix